### DISCUSSION TITLE
How to use scoped event loops as workaround for missing loop policy support?

### DISCUSSION THREAD

**Discussion by schlamar** [pytest-dev/pytest-asyncio#659] [General]
I don't see how I can use scoped event loops #620 as a workaround for my use case in #591

I want to define a loop policy for multiple tests divided in modules / packages / etc. But every test should run in a separate loop. So what I want is a class / module scoped event loop **policy**.

The only way to achieve this behavior is by defining a custom event_loop fixture AFAIS, which is now deprecated in 0.22. Or am I missing something?

**Comment by seifertm** [comment]
> The only way to achieve this behavior is by defining a custom event_loop fixture AFAIS, which is now deprecated in 0.22. Or am I missing something?

That's correct, yes.

Since there are other valid use cases that have been overlooked (cf #657), I'm considering yanking the v0.22 release or at least reverting the event_loop override deprecation, until there's sufficient non-deprecated functionality to replace the event loop overrides.

I originally created #646 to cover your use case. However, I can see that it can be impractical to mark every test case separately, especially when you have a lot of them. Do you have any suggestions how to define a scoped event loop policy that suits your needs?
